apps/src/craft/designer/craft.js

```diff
@@ -0,0 +1,913 @@
+  }
+
+  if (config.level.useScore) {
+    $('body').addClass("minecraft-scoring");
+  }
+
+  if (config.level.isEventLevel) {
+    $('body').addClass("minecraft-events");
+  }
+
+  var bodyElement = document.body;
+  bodyElement.className = bodyElement.className + " minecraft";
+
+  if (config.level.showPopupOnLoad) {
+    config.level.afterVideoBeforeInstructionsFn = (showInstructions) => {
+      var event = document.createEvent('Event');
+      event.initEvent('instructionsShown', true, true);
+      document.dispatchEvent(event);
+
+      if (config.level.showPopupOnLoad === 'playerSelection') {
+        Craft.showPlayerSelectionPopup(function (selectedPlayer) {
+          trackEvent('MinecraftDesigner', 'ChoseCharacter', selectedPlayer);
+          Craft.clearPlayerState();
+          trySetLocalStorageItem('craftSelectedPlayer', selectedPlayer);
+          Craft.updateUIForCharacter(selectedPlayer);
+          Craft.initializeAppLevel(config.level);
+          showInstructions();
+        });
+      }
+    };
+  }
+
+  if (config.level.puzzle_number && eventsLevelbuilderOverrides[config.level.puzzle_number]) {
+    Object.assign(config.level, eventsLevelbuilderOverrides[config.level.puzzle_number]);
+  }
+  Craft.initialConfig = config;
+
+  // replace studioApp methods with our own
+  studioApp.reset = this.reset.bind(this);
+  studioApp.runButtonClick = this.runButtonClick.bind(this);
+
+  Craft.level = config.level;
+  Craft.skin = config.skin;
+
+  var levelTracks = [];
+  if (Craft.level.songs && MUSIC_METADATA) {
+    levelTracks = MUSIC_METADATA.filter(function (trackMetadata) {
+      return Craft.level.songs.indexOf(trackMetadata.name) !== -1;
+    });
+  }
+
+  Craft.musicController = new MusicController(
+      studioApp.cdoSounds,
+      function (filename) {
+        return config.skin.assetUrl(`music/${filename}`);
+      },
+      levelTracks,
+      config.level.dayNightCycleTime ? 100 : (levelTracks.length > 1 ? 7500 : null)
+  );
+
+  // Play music when the instructions are shown
+  var playOnce = function () {
+    document.removeEventListener('instructionsHidden', playOnce);
+    if (studioApp.cdoSounds) {
+      studioApp.cdoSounds.whenAudioUnlocked(function () {
+        var hasSongInLevel = Craft.level.songs && Craft.level.songs.length > 1;
+        var songToPlayFirst = hasSongInLevel ? Craft.level.songs[0] : null;
+        Craft.musicController.play(songToPlayFirst);
+      });
+    }
+  };
+  document.addEventListener('instructionsHidden', playOnce);
+
+  const character = config.level.usePlayer ? eventsCharacters[Craft.getCurrentCharacter()] : eventsCharacters['Chicken'];
+
+  config.skin.staticAvatar = character.staticAvatar;
+  config.skin.smallStaticAvatar = character.smallStaticAvatar;
+  config.skin.failureAvatar = character.failureAvatar;
+  config.skin.winAvatar = character.winAvatar;
+
+  var levelConfig = config.level;
+
+  var onMount = function () {
+    studioApp.init(Object.assign({}, config, {
```

Comment:
nit: Instead of using Object.assign, you could now use spread syntax, i.e.
```
studioApp.init({
  ...config,
  forceInsertTopBlock: config.level.isEventLevel ? null : 'when_run'
  //etc
});